File: 2021/3/part_1.py
import numpy

with open("input.txt") as f:
    lines = [l for l in f.readlines() if l != "\n"]

g_rate = ""
e_rate = ""

for i in range(12):
    ones = 0
    zeros = 0
    for l in lines:
        if l[i] == "0":
            zeros += 1
        else:
            ones += 1
        
    if ones > zeros:
        g_rate += "1"
        e_rate += '0'
    else:
        g_rate += "0"
        e_rate += "1"

print(g_rate, e_rate)

print(eval(f"0b{g_rate}")*(eval(f"0b{e_rate}")))

File: 2021/3/part_1.js
const fs = require("fs");

const lines = fs.readFileSync("input.txt", "utf8").trim("\n").replace(/\r/g, "").split("\n");

let gRate = "";
let eRate = "";

for (i = 0; i < 12; i++) {
    let ones = 0;
    let zeros = 0;

    lines.forEach(l => {
        if (l[i] === "0") zeros += 1;
        else ones += 1;
    });

    if (ones > zeros) {
        gRate += "1";
        eRate += "0";
    } else {
        gRate += "0";
        eRate += "1";
    }
}

console.log(`gRate: ${gRate}, eRate: ${eRate}`);
console.log(parseInt(gRate, 2) * parseInt(eRate, 2));
